dev-proxy-abstractions/IProxyConfiguration.cs

```diff
@@ -21,4 +21,5 @@ public interface IProxyConfiguration
     IEnumerable<int> WatchPids { get; }
     IEnumerable<string> WatchProcessNames { get; }
     bool ShowTimestamps { get; }
+    long? Timeout { get; }
```

Comment:
In plugins, such as CachingGuidance or RateLimiting, we append the unit to the config option, to make it clearer to users what to expect, eg. `CacheTresholdSeconds`, `RetryAfterSeconds`, etc. I suggest we do it here too for consistency and clarity, ie. `TimeoutSeconds`. That way, users won't have to wonder what value to put in there.